crates/analytics/src/refunds/accumulator.rs

```diff
@@ -34,6 +71,93 @@ pub trait RefundMetricAccumulator {
     fn collect(self) -> Self::MetricOutput;
 }
 
+pub trait RefundDistributionAccumulator {
+    type DistributionOutput;
+
+    fn add_distribution_bucket(&mut self, distribution: &RefundDistributionRow);
+
+    fn collect(self) -> Self::DistributionOutput;
+}
+
+impl RefundDistributionAccumulator for RefundReasonDistributionAccumulator {
+    type DistributionOutput = Option<Vec<ReasonsResult>>;
+
+    fn add_distribution_bucket(&mut self, distribution: &RefundDistributionRow) {
+        self.refund_reason_vec.push(RefundReasonDistributionRow {
+            count: distribution.count.unwrap_or_default(),
+            total: distribution
+                .total
+                .clone()
+                .map(|i| i.to_i64().unwrap_or_default())
+                .unwrap_or_default(),
+            refund_reason: distribution.refund_reason.clone().unwrap_or("".to_string()),
+        })
+    }
+
+    fn collect(mut self) -> Self::DistributionOutput {
+        if self.refund_reason_vec.is_empty() {
+            None
+        } else {
+            self.refund_reason_vec.sort_by(|a, b| b.count.cmp(&a.count));
+            let mut res: Vec<ReasonsResult> = Vec::new();
+            for val in self.refund_reason_vec.into_iter() {
+                let perc = f64::from(u32::try_from(val.count).ok()?) * 100.0
+                    / f64::from(u32::try_from(val.total).ok()?);
+
+                res.push(ReasonsResult {
+                    reason: val.refund_reason,
+                    count: val.count,
+                    percentage: (perc * 100.0).round() / 100.0,
```

Comment:
Why the multiplication and division by 100?

If it's to only have 2 decimal places, be aware of the concerns with how floats are represented in computers, so the resulting number may be slightly off.

If you just need a string representation of this, you can use `format!()` (this even does the rounding):

```rust
format!("{perc:.02}");
```

Or if you need to perform further calculation, you can use a crate that handles decimals, such as the `rust_decimal` crate.

Reply:
The string representation is not required, need it to be an f64 itself.
Multiplication by 100, allows us to round it to the nearest integer, and again dividing by 100 will give us 2 decimal places, which is the desired behaviour.

Reply:
The reason why we'd have to be careful of doing floating point arithmetic like this is due to the gotchas involved: https://floating-point-gui.de/ (including a link in case you weren't aware already).

If we are completely sure that such arithmetic is safe, then we can proceed with it, no worries. Otherwise, we may have to consider handling it better.

Reply:
Understood, and I believe the arithmetic is safe. Thanks for the additional documentation link regarding this issue.